hudi-common/src/main/java/org/apache/hudi/common/table/TableSchemaResolver.java

```diff
@@ -212,10 +218,10 @@ public Schema getTableAvroSchemaWithoutMetadataFields(HoodieInstant instant) thr
    */
   private Option<Schema> getTableSchemaFromCommitMetadata(boolean includeMetadataFields) {
     HoodieTimeline timeline = metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants();
-    return getTableSchemaFromCommitMetadata(timeline.lastInstant().get(), includeMetadataFields);
+    return timeline.lastInstant().isPresent() ? getTableSchemaFromCommitMetadata(timeline.lastInstant().get(), includeMetadataFields) :
```

Comment:
would we move `timeline.lastInstant().isPresent()` inside `getTableSchemaFromCommitMetadata(HoodieInstant instant, boolean includeMetadataFields)`?